### PR TITLE
Assorted tsconfig fixes

### PR DESCRIPTION
**Pull request by ghost**
Some libraries are missing "dom" libs, others are missing path mappings.
Detected thanks to Microsoft/dtslint#145

### CODE REVIEW

File: types/ember__object/tsconfig.json [hunk 1]
```diff
@@ -15,8 +15,31 @@
...
+            "@ember/component/*": ["ember__component/*"],
+            "@ember/controller": ["ember__controller"],
+            "@ember/debug": ["ember__debug"],
+            "@ember/debug/*": ["ember__debug/*"],
+            "@ember/engine": ["ember__engine"],
+            "@ember/engine/*": ["ember__engine/*"],
+            "@ember/error": ["ember__error"],
             "@ember/object": ["ember__object"],
-            "@ember/object/*": ["ember__object/*"]
+            "@ember/object/*": ["ember__object/*"],
```
Comment (mike-north) [inline types/ember__object/tsconfig.json]: This doesn't make a whole lot of sense to me. `@ember/object` is a low-level package that `@ember/engine` depends on. My hope was to introducing all of these additional entanglements.

**Review by mike-north — CHANGES_REQUESTED** [review]

File: types/ember__object/tsconfig.json (hunk 1 above)
Comment (andy-ms) [reply types/ember__object/tsconfig.json]: The test files import 'ember' which brings in all these dependencies. An alternate fix would be to change the tests to test just this package, if you want to work on that.

Comment (mike-north) [reply types/ember__object/tsconfig.json]: proper fix is here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29761

**Review by mike-north — APPROVED** [review]
I would prefer #29761 merged before a release containing this PR is cut, so we don't get a `@types/ember__object` with extra dependencies.

**Comment by andy-ms** [comment]
@mike-north A bunch of other ember packages have similar problems, want to take a look at those too?

**Comment by mike-north** [comment]
@andy-ms I believe your fix is the correct fix for the rest of the ember-related packages. A search for `from 'ember';` in `types/ember__*` now shows no results.

**Pull request merged**